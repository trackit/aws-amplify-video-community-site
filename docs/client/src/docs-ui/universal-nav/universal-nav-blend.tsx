import * as links from "../../constants/links";
import {Component, h, Host, Prop} from "@stencil/core";
import {
  universalNavStyle,
  universalNavContentStyle,
  brandStyle,
  linksStyle,
  hideAboutLinkStyle,
  searchStyle,
} from "./universal-nav.style";

@Component({tag: "docs-universal-nav-blend", shadow: false})
export class DocsUniversalNavBlend {
  /*** what label should go next to the brand icon? */
  @Prop() readonly heading?: string;
  /*** image url for brand icon when nav in blend mode */
  @Prop() readonly brandIcon?: string;

  render() {
    return (
      <Host class={universalNavStyle}>
        <docs-container>
          <div class={universalNavContentStyle + " blend"}>
            <stencil-route-link
              url="/"
              anchorTitle="Amplify Docs"
              class={brandStyle}
            >
              <img src={this.brandIcon} />
              <span>{this.heading}</span>
            </stencil-route-link>

            <docs-search-bar class={searchStyle} />

            <div class={linksStyle}>
              <amplify-external-link
                redirect
                href={links.COMMUNITY}
                anchorTitle="Amplify Community"
              >
                <span>Community</span>
              </amplify-external-link>
              <amplify-external-link
                href={links.MARKETING}
                anchorTitle="AWS Amplify Homepage"
                graphic="black"
              >
                <span>
                  About<span class={hideAboutLinkStyle}>{` Amplify`}</span>
                </span>
              </amplify-external-link>
            </div>
          </div>
        </docs-container>
      </Host>
    );
  }
}
