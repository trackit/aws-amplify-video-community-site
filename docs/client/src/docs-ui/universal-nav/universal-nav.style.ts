import {css} from "emotion";
import {MQFablet, MQTablet, MQLaptop} from "../../amplify-ui/styles/media";

export const universalNavStyle = css`
  display: block;
  background-color: var(--bg-color);
`;

export const universalNavContentStyle = css`
  display: flex;
  flex: 1;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  transition: 0.25s ease all;
  padding: 1.5rem;
  flex-wrap: wrap;

  > * {
    flex: 1;
  }

  span {
    color: var(--color-white);
    white-space: nowrap;
  }

  &.blend {
    background-color: transparent;

    * {
      color: var(--font-color);
    }

    stencil-route-link > a > span {
      color: var(--color-orange-hv);
    }
  }
`;

export const brandStyle = css`
  order: 1;
  width: 50%;

  ${MQTablet} {
    margin-right: 3rem;
  }

  > a {
    display: flex;
    flex-direction: row;
    align-items: center;

    > img {
      height: 1rem;
      margin-right: 0.125rem;

      ${MQTablet} {
        height: 1.25rem;
      }
    }

    > span {
      margin-left: 0.25rem;
      font-size: 1rem;

      ${MQTablet} {
        font-size: 1.25rem;
      }
    }

    > sup {
      position: relative;
      top: -0.25rem;
      margin-left: 0.125rem;
      font-size: 0.5rem;
      font-weight: bold;
      color: var(--color-white);
    }
  }
`;

export const searchStyle = css`
  width: 100%;
  order: 3;
  margin-top: 1.5rem;
  flex: auto;
  ${MQTablet} {
    flex: 1 15rem;
    order: 2;
    margin: 0;
  }
`;

export const linksStyle = css`
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  order: 2;
  width: 50%;

  ${MQTablet} {
    margin-left: 3rem;
    order: 3;
  }

  a {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 0 1rem;
    font-size: 0.875rem;

    &:last-child {
      padding-right: 0;
    }
  }
`;

export const hideAboutLinkStyle = css`
  display: none;

  ${MQFablet} {
    display: initial;
  }
`;
