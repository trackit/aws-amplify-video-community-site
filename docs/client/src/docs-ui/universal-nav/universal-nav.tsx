import * as links from "../../constants/links";
import {Component, h, Host, Prop} from "@stencil/core";
import {
  universalNavStyle,
  universalNavContentStyle,
  brandStyle,
  linksStyle,
  hideAboutLinkStyle,
  searchStyle,
} from "./universal-nav.style";

@Component({tag: "docs-universal-nav", shadow: false})
export class DocsUniversalNav {
  /*** what label should go next to the brand icon? */
  @Prop() readonly heading?: string;
  /*** image url for brand icon */
  @Prop() readonly brandIcon?: string;

  render() {
    return (
      <Host class={universalNavStyle}>
        <docs-container class="background-color-orange-hv">
          <div class={universalNavContentStyle}>
            <stencil-route-link
              url="/"
              anchorTitle="Amplify Docs"
              class={brandStyle}
            >
              <img src={this.brandIcon} />
              <span>{this.heading}</span>
            </stencil-route-link>

            <docs-search-bar class={searchStyle} />

            <div class={linksStyle}>
              <amplify-external-link
                redirect
                href={links.COMMUNITY}
                anchorTitle="Amplify Community"
              >
                <span>Community</span>
              </amplify-external-link>
              <amplify-external-link
                href={links.MARKETING}
                anchorTitle="AWS Amplify Homepage"
                graphic={"white"}
              >
                <span>
                  About<span class={hideAboutLinkStyle}>{` Amplify`}</span>
                </span>
              </amplify-external-link>
            </div>
          </div>
        </docs-container>
      </Host>
    );
  }
}
