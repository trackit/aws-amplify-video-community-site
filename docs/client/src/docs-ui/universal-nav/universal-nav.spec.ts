import {DocsUniversalNav} from "./universal-nav";
import {newSpecPage} from "@stencil/core/testing";

describe("docs-universal-nav", () => {
  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [DocsUniversalNav],
            html: `<docs-universal-nav />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
