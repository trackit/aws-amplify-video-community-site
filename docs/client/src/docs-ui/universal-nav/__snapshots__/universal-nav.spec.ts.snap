// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs-universal-nav Render logic should render 1`] = `
<docs-universal-nav class="css-1c42ddj">
  <docs-container class="background-color-orange-hv">
    <div class="css-sx2vr3">
      <stencil-route-link anchortitle="Amplify Docs" class="css-11aid50" url="/">
        <img>
        <span></span>
      </stencil-route-link>
      <docs-search-bar class="css-fbz3bc"></docs-search-bar>
      <div class="css-14o8hhr">
        <amplify-external-link anchortitle="Amplify Community" href="https://amplify.aws/community" redirect="">
          <span>
            Community
          </span>
        </amplify-external-link>
        <amplify-external-link anchortitle="AWS Amplify Homepage" graphic="white" href="https://aws.amazon.com/amplify/framework/">
          <span>
            About
            <span class="css-ecbrz7">
              Amplify
            </span>
          </span>
        </amplify-external-link>
      </div>
    </div>
  </docs-container>
</docs-universal-nav>
`;
